core/src/test/java/cucumber/runtime/table/TableDifferTest.java

```diff
@@ -229,6 +230,41 @@ public void should_diff_with_empty_list() {
     }
 
     @Test(expected = TableDiffException.class)
+    public void should_diff_with_empty_table() {
+        try {
+            DataTable emptyTable = DataTable.create(Collections.emptyList());
+            table().diff(emptyTable);
+        } catch (IndexOutOfBoundsException e) {
```

Comment:
You are expecting the test to throw a TableDiffException but catching an IndexOutOfBoundsException. I don't think this is quite right.